File: C#/0219. Contains Duplicate II.cs
public class Solution {
    public bool ContainsNearbyDuplicate(int[] nums, int k) {
        Dictionary<int,int> numsDict=new Dictionary<int,int>();
        bool flag=false;
        for(int i=0;i<nums.Length;i++){
            if(!numsDict.ContainsKey(nums[i])){
                numsDict.Add(nums[i],i);
            }
            else{
                if(i-numsDict[nums[i]]<=k){
                    flag=true;
                    break;
                }
                else{
                    numsDict[nums[i]]=i;
                }
            }
        }
        return flag;
    }
}

File: archives/leetcode2/0219. Contains Duplicate II.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        dictnums={}
        for i in range(len(nums)):
            if nums[i] in dictnums:
                if i-dictnums[nums[i]]<=k:
                    return True
            dictnums[nums[i]]=i
        return False    
        
